src/redux/books/booksSlice.js

```diff
@@ -1,28 +1,55 @@
+/* eslint-disable no-param-reassign */
 import { createSlice } from '@reduxjs/toolkit';
-import bookStore from './bookShop';
+import { postBooks, fetchBooks, deleteBooks } from '../API/getAPI';
 
 const initialState = {
-  booklist: bookStore,
-  amount: bookStore.length,
-  total: 0,
+  booklist: [],
+  totalbooks: 0,
+  status: 'idle',
+  error: null,
+  createdStatus: '',
+  deletedStatus: '',
 };
 
 export const bookSlice = createSlice({
   name: 'book',
   initialState,
-  reducers: {
-    addBook: (state, action) => ({
+  extraReducers(builder) {
+    builder.addCase(fetchBooks.pending, (state) => ({
       ...state,
-      booklist: [...state.booklist, action.payload],
-    }),
-    removeBook: (state, action) => {
-      state.booklist.splice(state.booklist.findIndex(
-        (book) => book.id === action.payload,
-      ), 1);
-    },
+      status: 'loading',
+    }))
+      .addCase(fetchBooks.fulfilled, (state, action) => {
+        const keys = Object.keys(action.payload);
+        const temparray = [];
+        keys.forEach((key) => {
+          temparray.push(Object.assign({ id: key }, ...action.payload[key]));
+        });
+        state.booklist = [...temparray];
+        state.status = 'loaded';
+        state.totalbooks += 1;
+        // return {
+        //   ...state,
+        //   booklist: [...temparray],
+        //   status: 'loaded',
+        //   totalbooks: state.booklist.length + 0,
+        // };
```

Comment:
Kindly do not leave any unused codes as comments in your code base since it makes your project bulky, slow and does not conform to standard best practices. :wink: